File: scratch/records-tast-opportunity-20260916/full-scalar/src/Test/AckermannFFI.php
<?php
$exports['runAckermannFFI'] = function($m) {
    $ack = function($m, $n) use (&$ack) {
        if ($m === 0) return $n + 1;
        if ($m > 0 && $n === 0) return $ack($m - 1, 1);
        return $ack($m - 1, $ack($m, $n - 1));
    };
    return $ack((int)$m, 4);
};
return $exports;

File: scratch/purust-fbip-20260916/benchmark/src/Test/AckermannFFI.cc
#include "purescript.h"

namespace {
    int ackermann(int m, int n) {
        if (m == 0) return n + 1;
        if (n == 0) return ackermann(m - 1, 1);
        return ackermann(m - 1, ackermann(m, n - 1));
    }
}

FOREIGN_BEGIN(Test_AckermannFFI)
exports["runAckermannFFI"] = [](const boxed& in) -> boxed {
    const int result = ackermann(unbox<int>(in), 4);
    return result;
};
FOREIGN_END
